homeassistant/components/velbus/light.py

```diff
@@ -53,25 +73,47 @@ def brightness(self):
 
     def turn_on(self, **kwargs):
         """Instruct the Velbus light to turn on."""
-        try:
-            if ATTR_BRIGHTNESS in kwargs:
-                self._module.set_dimmer_state(
-                    self._channel,
-                    kwargs[ATTR_BRIGHTNESS],
-                    kwargs.get(ATTR_TRANSITION, 0),
-                )
+        if self._module.light_is_buttonled(self._channel):
+            if ATTR_FLASH in kwargs:
+                if kwargs[ATTR_FLASH] == FLASH_LONG:
+                    action = "slow"
+                elif kwargs[ATTR_FLASH] == FLASH_SHORT:
+                    action = "fast"
+                else:
+                    action = "on"
             else:
-                self._module.restore_dimmer_state(
-                    self._channel, kwargs.get(ATTR_TRANSITION, 0),
-                )
-        except VelbusException as err:
-            _LOGGER.error("A Velbus error occurred: %s", err)
+                action = "on"
+            try:
+                self._module.set_led_state(self._channel, action)
```

Comment:
We can make the same rewrite here as below.

Reply:
Done at this [commit](https://github.com/home-assistant/home-assistant/pull/30445/commits/a2993e677b3ec3f4c702adf2b7536c7bd23387fc)